client/src/components/ui/top-navtab.tsx

```diff
@@ -23,10 +24,20 @@ export default function TopNavtab({ tabs }: TopNavtabProps) {
         ? "border-penni-main text-penni-main"
         : "border-transparent text-penni-grey-inactive hover:text-penni-main focus:border-penni-main focus:text-penni-main"
     }`;
+  const targetDivRef = useRef<HTMLDivElement>(null);
+  const [paddingTop, setPaddingTop] = useState<number>(0);
 
+  useEffect(() => {
+    if (targetDivRef.current) {
+      setPaddingTop(targetDivRef.current.offsetHeight);
+    }
+  }, []);
   return (
-    <div className="flex w-full flex-col">
-      <div className="flex border-b">
+    <div className="relative flex w-full flex-col">
+      <div
+        ref={targetDivRef}
+        className={`${isFixed ? "fixed top-0" : ""} z-40 flex w-full border-b bg-white`}
```

Comment:
Should `bg-white` be `bg-penni-background-light-mode`?

Reply:
the latter is better, I reckon